File: app/src/main/java/com/example/canvastest/MainActivity.java
package com.example.canvastest;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

import engine.core.Engine;
import gameframework.assets.BulkLoader;
import engine.feature.TouchInputFeature;

public class MainActivity extends AppCompatActivity {
    private BulkLoader bulkLoader = new BulkLoader();

    private Engine engine;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);

        Engine.initStatic();

        bulkLoader.load(this);

        engine = new Engine(this);

        engine.enableFeature(TouchInputFeature.class);
        engine.setEnabledPerformanceOverlay(false);

        engine.boot(this, new BaloonsGame());
    }
}

File: app/src/main/java/life/unmedicated/jumpergame/MainActivity.kt
package life.unmedicated.jumpergame

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import life.unmedicated.engine2d.core.Engine
import life.unmedicated.engine2d.input.TouchInput

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val engine = Engine(this)

        TouchInput(engine)

        val game = JumperGame(engine)

        setContentView(engine.displayManager.displayView)

        engine.start()
    }
}